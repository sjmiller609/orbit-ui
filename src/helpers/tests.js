import { BrowserRouter } from 'react-router-dom'
import { shape } from 'prop-types'
import { mount } from 'enzyme'

// Instantiate router context
const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
}

const createContext = () => ({
  context: { router },
  childContextTypes: { router: shape({}) },
})

// wrap with router context - to mount RR links correctly
export function mountWrap(node) {
  return mount(node, createContext())
}
