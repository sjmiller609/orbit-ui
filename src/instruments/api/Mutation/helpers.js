const has = (str, find) => ~str.toLowerCase().indexOf(find.toLowerCase())

export const formErrors = ({ error, errors }) => {
  if (!error) return

  let err = JSON.stringify(error).toLowerCase()
  return errors.find(e => {
    if (e.key && has(err, e.key)) return e
    else if (e.allKeys && e.allKeys.every(k => has(err, k))) return e
    else if (e.someKeys && e.someKeys.some(k => has(err, k))) return e
  })
}
