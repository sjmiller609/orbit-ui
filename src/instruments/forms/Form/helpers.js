// faltten and stringify nested keys
export const unpack = (obj, ns, hash) => {
  if (!obj) return
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keypath = ns ? `${ns}.${key}` : key
    if (value && typeof value === 'object') {
      return unpack(value, keypath, acc)
    }
    acc[keypath] = value
    return acc
  }, hash || {})
}

const roll = (keys, obj, value) => {
  const key = keys[0]
  // case 1 - not obj
  if (keys.length === 1) {
    obj[key] = value
    return obj
  }
  // case 2 - doesn't exist
  if (!obj[key]) {
    obj[key] = isNaN(keys[1]) ? {} : []
  }
  // case 3 - exists
  obj[key] = roll(
    keys.slice(1),
    Array.isArray(obj[key]) ? [...obj[key]] : { ...obj[key] },
    value
  )

  return obj
}

// convert back into obj
export const pack = (obj, type = {}) => {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keys = key.split('.')

    return roll(keys, acc, value)
  }, type)
}

export const packChild = ({ name, obj }) => {
  const obj2 = {}
  let isArray
  Object.entries(obj).forEach(k => {
    if (name.length < k[0].length && k[0].indexOf(name) === 0) {
      const n = k[0].slice(name.length + 1)
      // must be true for all matching
      isArray = !isNaN(n.slice(0, n.indexOf('.')))
      obj2[n] = k[1]
    }
  })
  if (Object.keys(obj2).length) return pack(obj2, isArray ? [] : {})
  return null
}

export const removeChild = ({ name, obj = {} }) => {
  const obj2 = {}
  Object.entries(obj).forEach(k => {
    if (name.length >= k[0].length || k[0].indexOf(name) !== 0) {
      obj2[k[0]] = k[1]
    }
  })
  return obj2
}
