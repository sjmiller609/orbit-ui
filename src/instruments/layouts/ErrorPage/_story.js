import React from 'react'
import { storiesOf } from '@storybook/react'
import { ErrorPage } from 'instruments'

storiesOf('Instruments|Layouts', module).add('ErrorPage', () => (
  <ErrorPage>
    <div>[children]</div>
  </ErrorPage>
))
