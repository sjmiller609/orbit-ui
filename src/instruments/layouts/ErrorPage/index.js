'use strict'
import React from 'react'
import PropTypes from 'prop-types'

import { Page, Box, CardError, Logo } from 'instruments'
import classnames from 'classnames'

import s from './styles.scss'

const ErrorPage = ({ children, retry, className }) => {
  return (
    <Page
      className={classnames(s.page, className)}
      metaTitle="Error | Astronomer">
      <Box full className={s.noMatch}>
        <CardError retry={retry}>{children}</CardError>
        <Logo className={s.logo} noStars />
      </Box>
    </Page>
  )
}

ErrorPage.propTypes = {
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.element]),
  className: PropTypes.string,
  retry: PropTypes.func,
}

export default ErrorPage
