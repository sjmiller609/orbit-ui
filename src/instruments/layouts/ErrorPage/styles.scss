@import 'src/styles/mixins.scss';

img.logo {
  width: 10rem;
  margin-top: 5rem;
}
