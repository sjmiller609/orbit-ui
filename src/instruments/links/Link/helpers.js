// check if url is external
export const externalUrl = url => {
  if (typeof url === 'object') return false
  const path = /^https?:\/\/|^\/\//i
  if (path.test(url) && !~url.indexOf(window.location.origin)) return true
  return
}
