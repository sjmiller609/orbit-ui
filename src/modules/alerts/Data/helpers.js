import { formErrors } from 'instruments'

const errors = [
  {
    key: 'permissions',
    name: 'properties.alert_emails',
    error: 'Contact your workspace admin to upgrade your permissions.',
  },
]

export const handleError = error => formErrors({ error, errors })

export const trimError = error => error.split(':')[1].trim()
