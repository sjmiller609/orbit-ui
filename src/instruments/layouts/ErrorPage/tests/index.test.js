import React from 'react'
import { mountWrap } from 'helpers/tests'
import ErrorPage from '../index'

const props = {
  children: ['1', '2', '3'],
  className: 'testClassString',
  retry: jest.fn(),
}

describe('ErrorPage', () => {
  it('should be defined', () => {
    expect(ErrorPage).toBeDefined()
  })

  it('should render correctly', () => {
    const wrapper = mountWrap(<ErrorPage {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render different children', () => {
    const wrapper = mountWrap(
      <ErrorPage {...props} children={null}>
        <p>Child</p>
      </ErrorPage>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
