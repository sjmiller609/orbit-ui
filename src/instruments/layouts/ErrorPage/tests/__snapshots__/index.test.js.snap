// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage should render correctly 1`] = `
<ErrorPage
  className="testClassString"
  retry={[MockFunction]}
>
  <Page
    className="page testClassString"
    metaTitle="Error | Astronomer"
  >
    <div
      className="page page testClassString"
    >
      <Box
        className="noMatch"
        full={true}
      >
        <Flex
          className="noMatch"
          flow="box"
          full={true}
        >
          <div
            className="flex box undefined full noMatch"
          >
            <CardError
              retry={[MockFunction]}
            >
              <Card
                className="card"
              >
                <Box
                  className="card card"
                  justify="flex-start"
                >
                  <Flex
                    className="card card"
                    flow="box"
                    justify="flex-start"
                  >
                    <div
                      className="flex box flex-start card card"
                    >
                      <main
                        className="full"
                      >
                        <Block1
                          left={
                            <Imported
                              className="img"
                            />
                          }
                          title="DAG gone it! Something went wrong"
                        >
                          <Row
                            className="content"
                            wrap={true}
                          >
                            <Flex
                              className="content"
                              flow="row"
                              wrap={true}
                            >
                              <div
                                className="flex row undefined wrap content"
                              >
                                <Box
                                  className="left"
                                >
                                  <Flex
                                    className="left"
                                    flow="box"
                                  >
                                    <div
                                      className="flex box undefined left"
                                    >
                                      <Imported
                                        className="img"
                                      >
                                        <ReactImportedComponent
                                          className="img"
                                          loadable={
                                            Object {
                                              "done": true,
                                              "importFunction": [Function],
                                              "load": [Function],
                                              "mark": Array [],
                                              "ok": true,
                                              "payload": Object {
                                                "default": Object {
                                                  "process": [Function],
                                                },
                                                "process": [Function],
                                              },
                                              "promise": Promise {},
                                              "reset": [Function],
                                            }
                                          }
                                          render={[Function]}
                                        >
                                          <UnconnectedReactImportedComponent
                                            async={false}
                                            className="img"
                                            exportPicker={[Function]}
                                            loadable={
                                              Object {
                                                "done": true,
                                                "importFunction": [Function],
                                                "load": [Function],
                                                "mark": Array [],
                                                "ok": true,
                                                "payload": Object {
                                                  "default": Object {
                                                    "process": [Function],
                                                  },
                                                  "process": [Function],
                                                },
                                                "promise": Promise {},
                                                "reset": [Function],
                                              }
                                            }
                                            render={[Function]}
                                            streamId={0}
                                          >
                                            <img
                                              className="img"
                                              src={
                                                Object {
                                                  "process": [Function],
                                                }
                                              }
                                            />
                                          </UnconnectedReactImportedComponent>
                                        </ReactImportedComponent>
                                      </Imported>
                                    </div>
                                  </Flex>
                                </Box>
                                <Box
                                  align="flex-start"
                                  className="right"
                                >
                                  <Flex
                                    align="flex-start"
                                    className="right"
                                    flow="box"
                                  >
                                    <div
                                      className="flex box undefined align-flex-start right"
                                    >
                                      <H1>
                                        <h1
                                          className="h1"
                                        >
                                          DAG gone it! Something went wrong
                                        </h1>
                                      </H1>
                                      1
                                      2
                                      3
                                      <P>
                                        <p
                                          className="p"
                                        >
                                          <Link
                                            onClick={[MockFunction]}
                                          >
                                            <A
                                              className="link"
                                              onClick={[MockFunction]}
                                            >
                                              <a
                                                className="link"
                                                onClick={[MockFunction]}
                                              >
                                                Refresh
                                              </a>
                                            </A>
                                          </Link>
                                           to try again. Or if the problem persists, pleaseÂ 
                                          <Link
                                            to="https://www.astronomer.io/contact"
                                          >
                                            <A
                                              className="link"
                                              to="https://www.astronomer.io/contact"
                                            >
                                              <a
                                                className="link"
                                                href="https://www.astronomer.io/contact"
                                                target="_blank"
                                              >
                                                contact support
                                              </a>
                                            </A>
                                          </Link>
                                          .
                                        </p>
                                      </P>
                                    </div>
                                  </Flex>
                                </Box>
                              </div>
                            </Flex>
                          </Row>
                        </Block1>
                      </main>
                    </div>
                  </Flex>
                </Box>
              </Card>
            </CardError>
            <withRouter(Logo)
              className="logo"
              noStars={true}
            >
              <Route>
                <Logo
                  className="logo"
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={Object {}}
                  match={Object {}}
                  noStars={true}
                >
                  <Link
                    onClick={[Function]}
                  >
                    <A
                      className="link"
                      onClick={[Function]}
                    >
                      <a
                        className="link"
                        onClick={[Function]}
                      >
                        <Imported
                          className="logo stars logo"
                          title="Astronomer"
                        >
                          <ReactImportedComponent
                            className="logo stars logo"
                            loadable={
                              Object {
                                "done": true,
                                "importFunction": [Function],
                                "load": [Function],
                                "mark": Array [],
                                "ok": true,
                                "payload": Object {
                                  "default": Object {
                                    "process": [Function],
                                  },
                                  "process": [Function],
                                },
                                "promise": Promise {},
                                "reset": [Function],
                              }
                            }
                            render={[Function]}
                            title="Astronomer"
                          >
                            <UnconnectedReactImportedComponent
                              async={false}
                              className="logo stars logo"
                              exportPicker={[Function]}
                              loadable={
                                Object {
                                  "done": true,
                                  "importFunction": [Function],
                                  "load": [Function],
                                  "mark": Array [],
                                  "ok": true,
                                  "payload": Object {
                                    "default": Object {
                                      "process": [Function],
                                    },
                                    "process": [Function],
                                  },
                                  "promise": Promise {},
                                  "reset": [Function],
                                }
                              }
                              render={[Function]}
                              streamId={0}
                              title="Astronomer"
                            >
                              <img
                                className="logo stars logo"
                                src={
                                  Object {
                                    "process": [Function],
                                  }
                                }
                                title="Astronomer"
                              />
                            </UnconnectedReactImportedComponent>
                          </ReactImportedComponent>
                        </Imported>
                      </a>
                    </A>
                  </Link>
                </Logo>
              </Route>
            </withRouter(Logo)>
          </div>
        </Flex>
      </Box>
      <HelmetWrapper
        defer={true}
        encodeSpecialCharacters={true}
      >
        <SideEffect(NullComponent)
          defer={true}
          encodeSpecialCharacters={true}
          title="Error | Astronomer"
          titleAttributes={Object {}}
        >
          <NullComponent
            defer={true}
            encodeSpecialCharacters={true}
            title="Error | Astronomer"
            titleAttributes={Object {}}
          />
        </SideEffect(NullComponent)>
      </HelmetWrapper>
    </div>
  </Page>
</ErrorPage>
`;

exports[`ErrorPage should render different children 1`] = `
<ErrorPage
  className="testClassString"
  retry={[MockFunction]}
>
  <Page
    className="page testClassString"
    metaTitle="Error | Astronomer"
  >
    <div
      className="page page testClassString"
    >
      <Box
        className="noMatch"
        full={true}
      >
        <Flex
          className="noMatch"
          flow="box"
          full={true}
        >
          <div
            className="flex box undefined full noMatch"
          >
            <CardError
              retry={[MockFunction]}
            >
              <Card
                className="card"
              >
                <Box
                  className="card card"
                  justify="flex-start"
                >
                  <Flex
                    className="card card"
                    flow="box"
                    justify="flex-start"
                  >
                    <div
                      className="flex box flex-start card card"
                    >
                      <main
                        className="full"
                      >
                        <Block1
                          left={
                            <Imported
                              className="img"
                            />
                          }
                          title="DAG gone it! Something went wrong"
                        >
                          <Row
                            className="content"
                            wrap={true}
                          >
                            <Flex
                              className="content"
                              flow="row"
                              wrap={true}
                            >
                              <div
                                className="flex row undefined wrap content"
                              >
                                <Box
                                  className="left"
                                >
                                  <Flex
                                    className="left"
                                    flow="box"
                                  >
                                    <div
                                      className="flex box undefined left"
                                    >
                                      <Imported
                                        className="img"
                                      >
                                        <ReactImportedComponent
                                          className="img"
                                          loadable={
                                            Object {
                                              "done": true,
                                              "importFunction": [Function],
                                              "load": [Function],
                                              "mark": Array [],
                                              "ok": true,
                                              "payload": Object {
                                                "default": Object {
                                                  "process": [Function],
                                                },
                                                "process": [Function],
                                              },
                                              "promise": Promise {},
                                              "reset": [Function],
                                            }
                                          }
                                          render={[Function]}
                                        >
                                          <UnconnectedReactImportedComponent
                                            async={false}
                                            className="img"
                                            exportPicker={[Function]}
                                            loadable={
                                              Object {
                                                "done": true,
                                                "importFunction": [Function],
                                                "load": [Function],
                                                "mark": Array [],
                                                "ok": true,
                                                "payload": Object {
                                                  "default": Object {
                                                    "process": [Function],
                                                  },
                                                  "process": [Function],
                                                },
                                                "promise": Promise {},
                                                "reset": [Function],
                                              }
                                            }
                                            render={[Function]}
                                            streamId={0}
                                          >
                                            <img
                                              className="img"
                                              src={
                                                Object {
                                                  "process": [Function],
                                                }
                                              }
                                            />
                                          </UnconnectedReactImportedComponent>
                                        </ReactImportedComponent>
                                      </Imported>
                                    </div>
                                  </Flex>
                                </Box>
                                <Box
                                  align="flex-start"
                                  className="right"
                                >
                                  <Flex
                                    align="flex-start"
                                    className="right"
                                    flow="box"
                                  >
                                    <div
                                      className="flex box undefined align-flex-start right"
                                    >
                                      <H1>
                                        <h1
                                          className="h1"
                                        >
                                          DAG gone it! Something went wrong
                                        </h1>
                                      </H1>
                                      <p>
                                        Child
                                      </p>
                                      <P>
                                        <p
                                          className="p"
                                        >
                                          <Link
                                            onClick={[MockFunction]}
                                          >
                                            <A
                                              className="link"
                                              onClick={[MockFunction]}
                                            >
                                              <a
                                                className="link"
                                                onClick={[MockFunction]}
                                              >
                                                Refresh
                                              </a>
                                            </A>
                                          </Link>
                                           to try again. Or if the problem persists, pleaseÂ 
                                          <Link
                                            to="https://www.astronomer.io/contact"
                                          >
                                            <A
                                              className="link"
                                              to="https://www.astronomer.io/contact"
                                            >
                                              <a
                                                className="link"
                                                href="https://www.astronomer.io/contact"
                                                target="_blank"
                                              >
                                                contact support
                                              </a>
                                            </A>
                                          </Link>
                                          .
                                        </p>
                                      </P>
                                    </div>
                                  </Flex>
                                </Box>
                              </div>
                            </Flex>
                          </Row>
                        </Block1>
                      </main>
                    </div>
                  </Flex>
                </Box>
              </Card>
            </CardError>
            <withRouter(Logo)
              className="logo"
              noStars={true}
            >
              <Route>
                <Logo
                  className="logo"
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={Object {}}
                  match={Object {}}
                  noStars={true}
                >
                  <Link
                    onClick={[Function]}
                  >
                    <A
                      className="link"
                      onClick={[Function]}
                    >
                      <a
                        className="link"
                        onClick={[Function]}
                      >
                        <Imported
                          className="logo stars logo"
                          title="Astronomer"
                        >
                          <ReactImportedComponent
                            className="logo stars logo"
                            loadable={
                              Object {
                                "done": true,
                                "importFunction": [Function],
                                "load": [Function],
                                "mark": Array [],
                                "ok": true,
                                "payload": Object {
                                  "default": Object {
                                    "process": [Function],
                                  },
                                  "process": [Function],
                                },
                                "promise": Promise {},
                                "reset": [Function],
                              }
                            }
                            render={[Function]}
                            title="Astronomer"
                          >
                            <UnconnectedReactImportedComponent
                              async={false}
                              className="logo stars logo"
                              exportPicker={[Function]}
                              loadable={
                                Object {
                                  "done": true,
                                  "importFunction": [Function],
                                  "load": [Function],
                                  "mark": Array [],
                                  "ok": true,
                                  "payload": Object {
                                    "default": Object {
                                      "process": [Function],
                                    },
                                    "process": [Function],
                                  },
                                  "promise": Promise {},
                                  "reset": [Function],
                                }
                              }
                              render={[Function]}
                              streamId={0}
                              title="Astronomer"
                            >
                              <img
                                className="logo stars logo"
                                src={
                                  Object {
                                    "process": [Function],
                                  }
                                }
                                title="Astronomer"
                              />
                            </UnconnectedReactImportedComponent>
                          </ReactImportedComponent>
                        </Imported>
                      </a>
                    </A>
                  </Link>
                </Logo>
              </Route>
            </withRouter(Logo)>
          </div>
        </Flex>
      </Box>
      <HelmetWrapper
        defer={true}
        encodeSpecialCharacters={true}
      >
        <SideEffect(NullComponent)
          defer={true}
          encodeSpecialCharacters={true}
          title="Error | Astronomer"
          titleAttributes={Object {}}
        >
          <NullComponent
            defer={true}
            encodeSpecialCharacters={true}
            title="Error | Astronomer"
            titleAttributes={Object {}}
          />
        </SideEffect(NullComponent)>
      </HelmetWrapper>
    </div>
  </Page>
</ErrorPage>
`;
